test/unittests/value_test.cpp

```diff
@@ -13,11 +13,38 @@ TEST(value, value_initialization)
     EXPECT_EQ(v.i64, 0);
 }
 
+TEST(value, constructor_from_i32)
+{
+    Value v = int32_t{1};
+    EXPECT_EQ(v.i32, 1);
+    v = int32_t{-2};
+    EXPECT_EQ(v.i32, 0xfffffffe);
+    v = uint32_t{111};
+    EXPECT_EQ(v.i32, 111);
+
+    constexpr auto max_i32 = std::numeric_limits<uint32_t>::max();
+    v = max_i32;
+    EXPECT_EQ(v.i32, max_i32);
+}
+
+TEST(value, constructor_from_int)
+{
+    // The int type is likely the int32_t.
+    Value v = 0;
+    EXPECT_EQ(v.i32, 0);
+    v = -3;
+    EXPECT_EQ(v.i32, 0xfffffffd);
+
+    constexpr auto max_int = std::numeric_limits<uint32_t>::max();
```

Comment:
This is the same as in the test above, did you mean:
```suggestion
    constexpr auto max_int = std::numeric_limits<int>::max();
```